test/test_other.py

```diff
@@ -15028,3 +15028,6 @@ def test_extra_struct_info(self):
     stderr = self.run_process([EMCC, test_file('hello_world.c'), '--js-library', test_file('other/test_extra_struct_info.js')], stderr=PIPE).stderr
     self.assertContained('(before) AF_INET=2', stderr)
     self.assertContained('(after) AF_INET=42', stderr)
+
+  def test_fs_writev_partial_write(self):
```

Comment:
```suggestion
  @also_with_wasmfs
  def test_fs_writev_partial_write(self):
```
This will also test in WasmFS. The fix doesn't affect it, but I think it's worth verifying that we don't have this bug there (some of the WasmFS code was ported from JS).

Reply:
I added this, but how do I run the test with wasmfs? This is what I have been doing to run the test until now: `test/runner other.test_fs_writev_partial_write`

Reply:
`test/runner other.test_fs_writev_partial_write_wasmfs` should work, that is, that annotation adds a new test like the original but with suffix `_wasmfs`.

Reply:
I have now figured it out: `test/runner other.test_fs_writev_partial_write_wasmfs`

However I see that the test fails with `FS.makedev is not a function` error ([link to circleci](https://app.circleci.com/pipelines/github/emscripten-core/emscripten/35922/workflows/c4064c8f-7e08-4abe-8568-7af99a0e7d75/jobs/802019?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary&invite=true#step-111-377742_39))

So I added `FORCE_FILESYSTEM` back in and it leads to a new error `UNREACHABLE executed`

I am trying to debug that now.

Reply:
The unreachable is here: https://github.com/emscripten-core/emscripten/blob/8fdaf16b7d84537d441a469e3ab08471c5f0265e/system/lib/wasmfs/js_impl_backend.h#L104

with this backtrace:

```
RuntimeError: unreachable
    at test_writev_partial_write.wasm.wasmfs::handle_unreachable(char const*, char const*, unsigned int) (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[409]:0xffbe)
    at test_writev_partial_write.wasm.wasmfs::JSImplFile::setSize(long long) (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[636]:0x1b6b0)
    at test_writev_partial_write.wasm.wasmfs::DataFile::Handle::setSize(long long) (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[438]:0x11ff6)
    at test_writev_partial_write.wasm.doOpen(wasmfs::path::ParsedParent, int, unsigned int, wasmfs::Backend*, OpenReturnMode) (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[433]:0x11cb1)
    at test_writev_partial_write.wasm.__syscall_openat (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[439]:0x12095)
    at test_writev_partial_write.wasm.fopen (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[44]:0x1a0c)
    at test_writev_partial_write.wasm.test_via_stdio (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[26]:0x10ce)
    at test_writev_partial_write.wasm.__original_main (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[27]:0x11c0)
    at test_writev_partial_write.wasm.main (wasm://wasm/test_writev_partial_write.wasm-00a47e2e:wasm-function[28]:0x11ee)
    at /scratch/emscripten/emscripten/test_writev_partial_write.js:639:12
```

Reply:
(It feels like I should be opening another issue on this, but I continue for here in the meantime)

The unreachable is because we are trying to truncate the file (I `fopen`ed it in `"w"` mode). If I open in `"a"` mode so we don't truncate it works fine.

The truncating happens here:

https://github.com/emscripten-core/emscripten/blob/d72d7226f4733af8ff993dec70198cf09a24142d/system/lib/wasmfs/syscalls.cpp#L563-L564

New/additional commit coming in a moment with this change.

Reply:
> `test/runner other.test_fs_writev_partial_write_wasmfs` should work, that is, that annotation adds a new test like the original but with suffix `_wasmfs`.

Somehow I blanked on seeing this comment - much appreciated and sorry for overlooking it!

Reply:
> This will also test in WasmFS. The fix doesn't affect it, but I think it's worth verifying that we don't have this bug there (some of the WasmFS code was ported from JS).

FWIW the test passes on WasmFS because this case was fixed in WasmFS as part of https://github.com/emscripten-core/emscripten/pull/16895

Specifically the same condition I added exists here:

https://github.com/emscripten-core/emscripten/blob/d72d7226f4733af8ff993dec70198cf09a24142d/system/lib/wasmfs/syscalls.cpp#L159-L164

(I have a PR in a second that fixes the copy-paste error in the comment)